packages/shared/lib/clients/orchestrator.ts

```diff
@@ -0,0 +1,406 @@
+    createActivityLogMessageAndEnd,
+    updateSuccess as updateSuccessActivityLog
+} from '../services/activity/activity.service.js';
+import { SYNC_TASK_QUEUE, WEBHOOK_TASK_QUEUE } from '../constants.js';
+import { v4 as uuid } from 'uuid';
+import featureFlags from '../utils/featureflags.js';
+import errorManager, { ErrorSourceEnum } from '../utils/error.manager.js';
+import type { Config as ProviderConfig } from '../models/Provider.js';
+import type { LogLevel } from '@nangohq/types';
+import SyncClient from './sync.client.js';
+import type { Client as TemporalClient } from '@temporalio/client';
+import { LogActionEnum } from '../models/Activity.js';
+import type { TExecuteReturn, TExecuteProps } from '@nangohq/nango-orchestrator';
+
+const logger = getLogger('orchestrator');
+
+async function getTemporal(): Promise<TemporalClient> {
+    const instance = await SyncClient.getInstance();
+    if (!instance) {
+        throw new Error('Temporal client not initialized');
+    }
+    return instance.getClient() as TemporalClient;
+}
+
+export interface OrchestratorClientInterface {
+    execute(props: TExecuteProps): Promise<TExecuteReturn>;
+}
+
+export class Orchestrator {
+    private client: OrchestratorClientInterface;
+
+    public constructor(client: OrchestratorClientInterface) {
+        this.client = client;
+    }
+
+    async triggerAction<T = any>({
+        connection,
+        actionName,
+        input,
+        activityLogId,
+        environment_id,
+        writeLogs = true,
+        logCtx
+    }: {
+        connection: NangoConnection;
+        actionName: string;
+        input: object;
+        activityLogId: number;
+        environment_id: number;
+        writeLogs?: boolean;
+        logCtx: LogContext;
+    }): Promise<Result<T, NangoError>> {
+        const startTime = Date.now();
+        const workflowId = `${SYNC_TASK_QUEUE}.ACTION:${actionName}.${connection.connection_id}.${uuid()}`;
+        try {
+            if (writeLogs) {
+                await createActivityLogMessage({
+                    level: 'info',
+                    environment_id,
+                    activity_log_id: activityLogId,
+                    content: `Starting action workflow ${workflowId} in the task queue: ${SYNC_TASK_QUEUE}`,
+                    params: {
+                        input: JSON.stringify(input, null, 2)
+                    },
+                    timestamp: Date.now()
+                });
+                await logCtx.info(`Starting action workflow ${workflowId} in the task queue: ${SYNC_TASK_QUEUE}`, { input: JSON.stringify(input, null, 2) });
+            }
+
+            // Execute dry-mode: no await for now
+            const groupKey: string = 'action';
+            this.dryExecute({
+                executionId: `${groupKey}:account:${connection.account_id}:environment:${connection.environment_id}:connection:${connection.id}:action:${actionName}:at:${new Date().toISOString()}:${uuid()}`,
```

Comment:
connection does not contain account_id, triggerAction actually receive Connection not NangoConnection

Reply:
nb: I have added a @deprecated mention on my branch to avoid further issue since I got bite by this too